git-annex-remote-rclone

```diff
@@ -90,18 +90,24 @@ ask () {
 	fi
 }
 
-GREP () {
-	set +e
-	out=$(grep "$@")
-	rc=$?
-	set -e
+printcmdresult() {
+	cmd="$1"
+	rc="$2"
+	out="$3"
 	# Replace explicit newline since we must provide 1 line DEBUG
 	# Fancy sed is from Example 5 of https://linuxhint.com/newline_replace_sed
 	# which worked on Linux and OSX.
 	# shellcheck disable=SC2016
 	out_safe=$(echo "$out" | sed -ne 'H;${x;s/\n/\\n/g;s/^,//;p;}')
-	echo "DEBUG 'grep \"$*\"' exited with rc=$rc and stdout=${out_safe}"
-	exit $rc
+	echo "DEBUG '$cmd' exited with rc=$rc and stdout=${out_safe}"
+}
+
+GREP () {
+	set +e
+	out=$(grep "$@")
+	rc=$?
+	set -e
+	printcmdresult "grep \"$*\"" "$rc" "$out"
```

Comment:
```suggestion
	printcmdresult "grep \"$*\"" "$rc" "$out"
	exit $rc
```

was it forgotten or abandoned? (I permitted tests running for this PR just now -- if they pass, not good since would mean that we do not test for correct operation anyhow here)

Reply:
I forgot it. `GREP` was actually only used in one place, in the implementation of `REMOVE`:

```
if remove_result=$(rclone delete --retries 1 "$LOC$key" 2>&1); then
    echo REMOVE-SUCCESS "$key"
else
    if echo "$remove_result" | GREP ' directory not found'; then
        echo REMOVE-SUCCESS "$key"
    else
        echo REMOVE-FAILURE "$key"
    fi
fi
```

The `REMOVE-FAILURE` branch is actually not covered by the tests, because I don't know how to trigger that without mocking `rclone`. Therefor the missing `exit` didn't trigger a test failure.

I've added `return $rc` instead of `exit $rc`. The effect within an `if` is the same, but `exit` would be very confusing if the function were used in a different way in the future.

Reply:
great, thank you for the explanation and `exit -> return`